Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+
+### File Tags
+
+_File tags_ are file level metadata that is relevant for all nodes in the file.
+A common use for file tags is for versioning the file.
+File tags must go at the start of a file before any nodes begin.
+File tags must have the `#` symbol at the start of them but are syntactically identical otherwise to the Header Tags of nodes.
+
+### Nodes
+
+A *node* is the single story element of a yarn file.
+Nodes are the story structural building blocks for yarn.
+Nodes are designed to contain pieces of a story and then have these story pieces linked together.
+This is not a requirement, everything could be done in a single node, this would just be unwieldy.
+A node must be comprised of a single header and a single body in that order.
+
+### Headers
+
+A header is comprised of one or more header tags.
+The header is finished when encountering a line that only contains the *header delimiter* `---`.
+After encountering the header delimiter the body of the node is entered.
+
+#### Header Tags
+
+A *header tag* is a line broken up into three components, in order; the tag name, the separator, and the tag text.
+The *tag name* can be any combination of the characters `A` - `Z`, `a` - `z`, `0` - `9`.
+The *tag separator* is the character `:`.
+The *tag text* can be any combination of the characters `A` - `Z`, `a` - `z`, `0` - `9`.
+Header tags are commonly used as node specific metadata but using them in this manner is not required.
+
+The amount of allowed whitespace between the tag name, the seperator, and the tag text is unspecified.
+An example of a header tag is the title tag: `Title:start`.
+
+Every node must have a title tag.
+Required or banned header tags beyond title are unspecified.
+The order of header tags is unspecified.
+
+#### Title Tag
+
+The *title tag* is a specific header tag that uniquely identifies the node.
+The tag name for the title tag must be `Title`.
+The tag text for the title tag must be unique within the file.
+The tag text for the title tag should be unique within the project.
+
+The behaviour of the program when a title tag's text is not unique across the project is unspecified.
+The program could either flag this as an error and abort, choose which node will be the canonical one with that title, or some other form of name conflict resolution.
+
+### Body
+
+A body is comprised of multiple statements.
+A *statement* is a line that is one of the following:
+
+- dialogue
+- commands
+- options
+
+A statement may have optional hashtags.
+A body must have at least one statement.
+
+The body ends when encountering a line that contains the *body delimter* `===`.
+The body delimiter ends both the current node and the body of that node.
+The end of file must not be used in place of a body delimiter.
+
+#### Hashtags
+
+*Hashtags* are metadata associated with the statement they are a part of.
+Hashtags must go at the end of the statement.
```

Comment:
How should hashtags interact with comments i.e. which of the following examples allows for a hashtag to be recognised: `Statement \\Comment #Hashtag` vs. `Statement #Hashtag \\Comment`

Reply:
comments take all symbols from the terminal character to the newline symbol as described in their section.
As such the `#` is meaningless, no different than any other character